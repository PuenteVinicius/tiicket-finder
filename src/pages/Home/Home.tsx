import React from "react";
import { Box, Main, Heading, Paragraph } from "grommet";

function Home() {
  return (
    <Box
      direction="row"
      pad="xlarge"
      align="center"
      alignContent="center"
      alignSelf="center"
    >
      <Main pad="large">
        <Heading alignSelf="center">Busque por uma banda ou artista</Heading>
        <Paragraph alignSelf="center">
          Iremos mostrar ingressos e videos do seu artista preferido :)
        </Paragraph>
      </Main>
    </Box>
  );
}

export default Home;
