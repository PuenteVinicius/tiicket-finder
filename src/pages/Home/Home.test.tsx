import { render } from "@testing-library/react";
import "jest-styled-components";
import Home from "./Home";
describe("Home component", () => {
  let view: any;

  beforeEach(() => {
    view = render(<Home />);
  });

  it("Should render text correctly", () => {
    expect(
      view.getByText("Busque por um artista")
    ).toBeInTheDocument();
    expect(
      view.getByText(
        "Iremos mostrar ingressos e videos do seu artista preferido :)"
      )
    ).toBeInTheDocument();
  });

  it("renders home correctly", () => {
    expect(view).toMatchSnapshot();
  });
});
