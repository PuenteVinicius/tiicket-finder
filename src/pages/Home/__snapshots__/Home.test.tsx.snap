// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders home correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 96px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  height: 100%;
  padding: 48px;
  overflow: auto;
}

.c2 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 600;
}

.c3 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  font-size: 18px;
  line-height: 24px;
  max-width: 432px;
}

@media only screen and (max-width:768px) {
  .c0 {
    padding: 48px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    padding: 24px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    font-size: 34px;
    line-height: 40px;
    max-width: 816px;
  }
}

<body>
    <div>
      <div
        class="c0"
      >
        <main
          class="c1"
        >
          <h1
            class="c2"
          >
            Busque por uma banda ou artista
          </h1>
          <p
            class="c3"
          >
            Iremos mostrar ingressos e videos do seu artista preferido :)
          </p>
        </main>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="StyledBox-sc-13pk1d4-0 fprswP"
    >
      <main
        class="StyledBox-sc-13pk1d4-0 llmstX"
      >
        <h1
          class="StyledHeading-sc-1rdh4aw-0 gMhqIv"
        >
          Busque por uma banda ou artista
        </h1>
        <p
          class="StyledParagraph-tbetod-0 dRiYAb"
        >
          Iremos mostrar ingressos e videos do seu artista preferido :)
        </p>
      </main>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
